File: src/IntroToKotlin/JavaPart/Location.java
package IntroToKotlin.JavaPart;

import java.util.HashMap;
import java.util.Map;

public class Location {
    private final Integer locationId;
    private final String description;
    private final Map<String, Integer> exits;

    public Location(Integer locationId, String description) {
        this.locationId = locationId;
        this.description = description;
        exits = new HashMap<>();
        this.exits.put("Q", 0);
    }

    public Integer getLocationId() {return locationId;}
    public String getDescription() {return description;}
    public Map<String, Integer> getExits() {return exits;}

    protected void addExit(String direction, Integer location) {exits.put(direction,location);}
}

File: src/IntroToKotlin/KotlinPart/Location.kt
package IntroToKotlin.KotlinPart

// DECLARAS LAS CONSTANTES O VARIABLES NECESARIAS EN LA CLASE Y SU TIPO DETRAS DE LOS :
data class Location (val locationId: Int, val description: String){
    val exits = mutableMapOf<String, Int>()

    init {
        exits["Q"] = 0
    }

    fun addExit(direction: String, destinationId: Int) {
        // ACCEDES A UN MAPA COMO UN OBJETO EN JS
        exits[direction] = destinationId
    }
}